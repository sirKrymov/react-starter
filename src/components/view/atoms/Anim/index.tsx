// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

interface IProps {
  duration?: string;
  children: ReactNode;
  delay?: string;
  type?: 'mount' | 'show';
}

export function Anim({
  children,
  duration = '0.4s',
  delay = '0.2s',
  type = 'show'
}: IProps): ReactElement {
  return (
    <div
      style={{
        animationDuration: duration,
        animationFillMode: 'both',
        animationDelay: delay,
        animationName: type,
        opacity: 0
      }}
    >
      {children}
    </div>
  );
}
