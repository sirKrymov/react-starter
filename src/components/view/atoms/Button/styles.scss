@import '../../../../styles/configs';

.button {
  border-radius: $border-radius;
  line-height: 1;
  transition: opacity $anim-time-fast, background-color $anim-time-fast,
    border-color $anim-time-fast, color $anim-time-fast;
  min-height: inherit;
  padding: 0 16px;
  width: 100%;

  &:active {
    opacity: 0.75;
  }

  &__content {
    align-items: center;
    display: inline-flex;
  }

  &--loading {
    pointer-events: none;
    cursor: default;

    &:hover {
      opacity: 1;
    }
  }

  &--color-secondary {
    background-color: var(--secondary);
    color: var(--secondaryContrastText);

    &.button--variant-outlined {
      background-color: transparent;
      border: 2px solid var(--secondary);
      color: var(--secondary);

      &:hover {
        background-color: var(--secondary);
        color: var(--secondaryContrastText);
      }
    }

    &.button--variant-contained {
      background-color: var(--secondary);
      box-shadow: $box-shadow-light;
      color: var(--secondaryContrastText);

      &:hover {
        background-color: var(--secondaryDark);
      }
    }
  }

  &--disabled {
    pointer-events: none;
    opacity: 0.6;
    cursor: not-allowed;
  }

  &--padding-xs {
    padding: 0 16px;
  }

  &--padding-sm {
    padding: 0 24px;
  }

  &--padding-md {
    padding: 0 32px;
  }

  &--padding-lg {
    padding: 0 40px;
  }

  &--font-weight-bold {
    font-weight: 700;
  }

  &--font-weight-default {
    font-weight: initial;
  }
}
