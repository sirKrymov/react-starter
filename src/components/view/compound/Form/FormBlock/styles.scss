.form-block {
  flex-direction: column;
  display: flex;
  width: 100%;

  &--mb-s1 {
    margin-bottom: 8px;
  }

  &--mb-s2 {
    margin-bottom: 16px;
  }

  &--mb-s3 {
    margin-bottom: 24px;
  }

  &--mb-s4 {
    margin-bottom: 32px;
  }

  &--mb-s5 {
    margin-bottom: 40px;
  }

  &--mb-s6 {
    margin-bottom: 48px;
  }

  &--mb-s7 {
    margin-bottom: 56px;
  }

  &--mb-s8 {
    margin-bottom: 64px;
  }

  &--mb-s9 {
    margin-bottom: 72px;
  }

  &--mb-s10 {
    margin-bottom: 80px;
  }

  &--align-center {
    align-items: center;
  }
}
