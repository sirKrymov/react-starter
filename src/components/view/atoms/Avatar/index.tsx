// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  withBorder?: boolean;
  className?: string;
  onClick?(): void;
  avatar?: string;
  size?: 'xs' | 'sm' | 'lg' | number;
  id: string;
}

export function Avatar({
  className,
  onClick,
  withBorder,
  avatar,
  size = 'xs',
  id
}: IProps): ReactElement {
  const avatarClass = cn(
    {
      avatar: true,
      'avatar--with-border': withBorder,
      'avatar--with-click': onClick,
      ...(typeof size === 'string' && { [`avatar--${size}`]: size })
    },
    className
  );

  return (
    <div
      className={avatarClass}
      onClick={onClick ? onClick : undefined}
      style={
        typeof size === 'number'
          ? {
              height: size,
              width: size
            }
          : {}
      }
      id={id}
    >
      <img
        className="img-cover"
        src={
          avatar ||
          'https://f0.pngfuel.com/png/980/886/male-portrait-avatar-png-clip-art.png'
        }
        alt="avatar"
      />
    </div>
  );
}
