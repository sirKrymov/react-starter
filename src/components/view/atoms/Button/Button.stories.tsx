import React from 'react';

import { jsxDecorator } from 'storybook-addon-jsx';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Button } from '../Button';

import '../Button/styles.scss';

storiesOf('Button', module)
  .addDecorator(jsxDecorator)
  .add('Simple text', () => {
    return (
      <Button onClick={action('Clicked')} id="button">
        Hello Button
      </Button>
    );
  });
