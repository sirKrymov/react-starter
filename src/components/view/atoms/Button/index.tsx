// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { CSSProperties, ReactElement, ReactNode, memo } from 'react';

import cn from 'classnames';

import { Typography } from 'components/view/atoms/Typography';
import { Loader } from 'components/view/atoms/loaders/Loader';
import { UiSize } from 'components/view/atoms/UiSize';

import './styles.scss';

interface IProps {
  fontWeight?: 'default' | 'bold';
  uppercase?: boolean;
  onClick?(): void | undefined;
  className?: string;
  disabled?: boolean;
  children: ReactNode;
  variant?: 'contained' | 'outlined';
  loading?: boolean;
  height?: 'xs' | 'sm' | 'md' | 'lg' | 'full';
  border?: boolean;
  width?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'full' | 'content';
  color?: 'secondary';
  style?: CSSProperties;
  type?: 'button' | 'submit';
  id: string;
}

function ButtonBase({
  fontWeight = 'bold',
  uppercase,
  className,
  disabled,
  children,
  onClick,
  variant = 'contained',
  loading,
  border,
  height,
  width,
  style,
  color = 'secondary',
  type = 'button',
  id
}: IProps): ReactElement {
  const buttonClass = cn(
    {
      button: true,
      [`button--font-weight-${fontWeight}`]: fontWeight,
      [`button--variant-${variant}`]: variant,
      [`button--padding-${height}`]: height && width === 'content',
      [`button--color-${color}`]: color,
      [`button--disabled`]: disabled,
      [`button--border`]: border,
      'button--loading': loading
    },
    className
  );

  return (
    <UiSize width={width} height={height}>
      <button
        className={buttonClass}
        disabled={disabled}
        onClick={loading || disabled ? undefined : onClick}
        style={style}
        type={type}
        id={id}
      >
        {loading ? (
          <Loader color="on-secondary" />
        ) : (
          <div className="button__content">
            <Typography uppercase={uppercase} variant="inherit" color="inherit">
              {children}
            </Typography>
          </div>
        )}
      </button>
    </UiSize>
  );
}

export const Button = memo(ButtonBase);
