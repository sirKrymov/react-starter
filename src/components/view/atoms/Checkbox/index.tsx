// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ChangeEvent, ReactElement } from 'react';

import { IoMdCheckmark } from 'react-icons/io';
import cn from 'classnames';

import './styles.scss';

interface IProps {
  defaultChecked?: boolean;
  onChange(e: ChangeEvent<any>): void;
  className?: string;
  disabled?: boolean;
  value: boolean;
  size?: string;
  name: string;
  id: string;
}

export function Checkbox({
  defaultChecked,
  className,
  onChange,
  disabled,
  value,
  size,
  name,
  id
}: IProps): ReactElement {
  const checkboxClass = cn(
    {
      checkbox: true,
      [`checkbox--size-${size}`]: size,
      'checkbox--disabled': disabled
    },
    className
  );

  return (
    <div className={checkboxClass}>
      <input
        defaultChecked={defaultChecked}
        onChange={onChange}
        disabled={disabled}
        checked={value}
        type="checkbox"
        name={name}
        id={id}
      />

      <label className="checkbox__label" htmlFor={id}>
        <span className="checkbox__icon">
          <IoMdCheckmark />
        </span>
      </label>
    </div>
  );
}
