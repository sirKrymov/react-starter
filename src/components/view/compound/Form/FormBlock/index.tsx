// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  marginBottom?:
    | 's1'
    | 's2'
    | 's3'
    | 's4'
    | 's5'
    | 's6'
    | 's7'
    | 's8'
    | 's9'
    | 's10';
  children: ReactNode;
  alignH?: 'center';
}

export function FormBlock({
  marginBottom,
  children,
  alignH
}: IProps): ReactElement {
  const formBlockClass = cn({
    'form-block': true,
    [`form-block--mb-${marginBottom}`]: marginBottom,
    [`form-block--align-${alignH}`]: alignH
  });

  return <div className={formBlockClass}>{children}</div>;
}
