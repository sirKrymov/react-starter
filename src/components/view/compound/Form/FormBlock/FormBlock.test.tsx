import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { FormBlock } from '../FormBlock';
import { Button } from '../../../atoms/Button';

describe('<FormBlock> compound component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(
      <FormBlock>
        <Button id="test">Test</Button>
      </FormBlock>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
