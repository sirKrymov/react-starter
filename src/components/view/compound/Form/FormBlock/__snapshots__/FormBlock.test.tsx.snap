// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormBlock> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="form-block"
>
  <Memo(ButtonBase)
    id="test"
  >
    Test
  </Memo(ButtonBase)>
</div>
`;
